include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -8121,7 +8045,9 @@ FEFaceEvaluation<dim,
     this->active_fe_index,
     this->dof_info);
 
-  if (fe_degree > 0 &&
+  if (this->data->data.front().fe_degree > 0 &&
+      fast_evaluation_supported(this->data->data.front().fe_degree,
+                                this->data->data.front().n_q_points_1d) &&
```

Comment:
This check is needed because else `FEFaceEvaluationImplGatherEvaluateSelector` does not work? Is it possible to add there an assert?

Reply:
Our current implementation does not work with the run-time version, and it did not on current master, where we escaped the whole optimized section in case we did not have `fe_degree>0`/`fe_degree>1`: https://github.com/dealii/dealii/blob/58c0c31147053395d69415a9fd2f8bb2735c38e6/include/deal.II/matrix_free/evaluation_kernels.h#L3523-L3533
I think it might be a different topic to enable this later, as the work to make it happen is not that different from the other places.